syntax = "proto3";

package gpssfdw;

message EstimateSizeRequest {
    string id = 1;
}

message EstimateSizeResponse {
    int64 estimate_size = 1;
}

message StreamDataRequest {
    string id = 1;
    int32 segid = 2;
}

message StreamDataResponse {
    bytes msg = 1;
}

service GpssFdw {
    rpc EstimateSize(EstimateSizeRequest) returns (EstimateSizeResponse);
    rpc StreamData(StreamDataRequest) returns (stream StreamDataResponse);
}